From e0bf8c152a76651a407209666bcdcb0dcb27ab5a Mon Sep 17 00:00:00 2001
From: Adrien Ricciardi <aricciardi@baylibre.com>
Date: Mon, 9 Oct 2023 12:41:07 +0200
Subject: [PATCH 29/38] net: can: Added extended CAN ID support.

Also updated the CAN demo driver and the CAN demo example accordingly.

Signed-off-by: Adrien Ricciardi <aricciardi@baylibre.com>
---
 examples/posix/can_demo/can_demo.c   | 4 ++--
 machines/posix/tpl_can_demo_driver.c | 2 +-
 net/can/Can_GeneralTypes.h           | 3 ++-
 3 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/examples/posix/can_demo/can_demo.c b/examples/posix/can_demo/can_demo.c
index e08d5beb..fb1eb138 100644
--- a/examples/posix/can_demo/can_demo.c
+++ b/examples/posix/can_demo/can_demo.c
@@ -121,7 +121,7 @@ TASK(can_task)
 
 		pointer_can_pdu = (Can_PduType *) pdu_info.SduDataPtr;
 		printf("ID = 0x%X, flags = 0x%02X, length = %d, payload = ",
-			pointer_can_pdu->id & TPL_CAN_ID_MASK,
+			pointer_can_pdu->id & TPL_CAN_ID_STANDARD_MASK,
 			TPL_CAN_ID_TYPE_GET(pointer_can_pdu->id),
 			pointer_can_pdu->length);
 		for (i = 0; i < pointer_can_pdu->length; i++)
@@ -151,7 +151,7 @@ TASK(can_task)
 
 		pointer_can_pdu = (Can_PduType *) pdu_info.SduDataPtr;
 		printf("ID = 0x%X, flags = 0x%02X, length = %d, payload = ",
-			pointer_can_pdu->id & TPL_CAN_ID_MASK,
+			pointer_can_pdu->id & TPL_CAN_ID_EXTENDED_MASK,
 			TPL_CAN_ID_TYPE_GET(pointer_can_pdu->id),
 			pointer_can_pdu->length);
 		for (i = 0; i < pointer_can_pdu->length; i++)
diff --git a/machines/posix/tpl_can_demo_driver.c b/machines/posix/tpl_can_demo_driver.c
index 2fc67603..64d418d2 100644
--- a/machines/posix/tpl_can_demo_driver.c
+++ b/machines/posix/tpl_can_demo_driver.c
@@ -109,7 +109,7 @@ static Std_ReturnType can_demo_driver_transmit(struct tpl_can_controller_t *ctrl
 		__func__,
 		__LINE__,
 		ctrl->base_address,
-		pdu_info->id & TPL_CAN_ID_MASK,
+		pdu_info->id & TPL_CAN_ID_EXTENDED_MASK,
 		TPL_CAN_ID_TYPE_GET(pdu_info->id),
 		pdu_info->length);
 	for (i = 0; i < pdu_info->length; i++)
diff --git a/net/can/Can_GeneralTypes.h b/net/can/Can_GeneralTypes.h
index 0885ef33..dfc37187 100644
--- a/net/can/Can_GeneralTypes.h
+++ b/net/can/Can_GeneralTypes.h
@@ -36,7 +36,8 @@
 #define TPL_CAN_ID_TYPE_FD_EXTENDED (0x03U << 30)
 #define TPL_CAN_ID_TYPE_MASK (0x03U << 30)
 #define TPL_CAN_ID_TYPE_GET(id) ((id & TPL_CAN_ID_TYPE_MASK) >> 30)
-#define TPL_CAN_ID_MASK (0x3FFFFFFFU)
+#define TPL_CAN_ID_STANDARD_MASK (0x3FFU)
+#define TPL_CAN_ID_EXTENDED_MASK (0x3FFFFFFFU)
 
 #define TPL_CAN_CLASSIC_FRAME_MAXIMUM_PAYLOAD_SIZE (8)
 #define TPL_CAN_FD_FRAME_MAXIMUM_PAYLOAD_SIZE (64)
-- 
2.34.1

